### PR TITLE
Update documentation and user guides for beta snapshotter

### LINKED ISSUE #457: Update deployments to include new snapshot controller (for v1beta1 APIs), start running tests, and update docs
Looks like some new snapshot APIs went Beta in k8s `1.17`. The corresponding `external-snapshotter` is also available and tested with the PD Driver. However, the stable deployment in this repo still does not contain the snapshotter, tests are not run by default, and the documentation for snapshotting may be out of date.

### PR DESCRIPTION
**Pull request by saikat-royc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Update the documentation and user guides for beta snapshotter feature.
**Which issue(s) this PR fixes**:

Fixes #457 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Update documentation and user guides for beta snapshotter
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ that represents availability by zone.
 | Snapshots       | Alpha | 1.13                          | Any                          | v0.3.0             | Alpha              |
 | Resize (Expand) | Alpha | 1.14                          | 1.14                         | v0.6.0             | Alpha              |
 | Resize (Expand) | Beta  | 1.16                          | 1.16                         | v0.7.0             | Stable             |
-
+| Snapshots       | Beta  | 1.17                          | 1.17                         | master             | Stable             |
```
Comment (msau42) [inline README.md:73]: Node version is "Any" since snapshots is only a control plane operation.

File: docs/kubernetes/user-guides/snapshots.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
-# Kubernetes Snapshots User Guide (Alpha)
+# Kubernetes Snapshots User Guide (Beta)
 
->**Attention:** VolumeSnapshot is an alpha feature. Make sure you have enabled it in Kubernetes API server using `--feature-gates=VolumeSnapshotDataSource=true` flag.
+>**Attention:** Attention: VolumeSnapshot is a Beta feature enabled by default in Kubernetes 1.17+. Attention: VolumeSnapshot is only available in the driver version "master".
 
-### Install Driver with alpha snapshot feature
+### Install Driver with beta snapshot feature
 
 1. [One-time per project] Create GCP service account for the CSI driver and set required roles
```
Comment (msau42) [inline docs/kubernetes/user-guides/snapshots.md:7]: We can link to the install driver user guide for this step

Comment (saikat-royc) [reply docs/kubernetes/user-guides/snapshots.md:7]: Ack

File: README.md (hunk 1 above)
Comment (saikat-royc) [reply README.md:73]: Thanks for pointing out.

File: docs/kubernetes/user-guides/snapshots.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
-# Kubernetes Snapshots User Guide (Alpha)
+# Kubernetes Snapshots User Guide (Beta)
 
->**Attention:** VolumeSnapshot is an alpha feature. Make sure you have enabled it in Kubernetes API server using `--feature-gates=VolumeSnapshotDataSource=true` flag.
+>**Attention:** Attention: VolumeSnapshot is a Beta feature enabled by default in Kubernetes 1.17+. Attention: VolumeSnapshot is only available in the driver version "master".
 
-### Install Driver with alpha snapshot feature
+### [Install Driver](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/docs/kubernetes/user-guides/driver-install.md) with beta snapshot feature.
 
 1. [One-time per project] Create GCP service account for the CSI driver and set required roles
```
Comment (msau42) [inline docs/kubernetes/user-guides/snapshots.md:7]: I think most of the steps in this section can be removed since it's covered by the driver-install.md

Comment (saikat-royc) [reply docs/kubernetes/user-guides/snapshots.md:7]: Ack

File: docs/kubernetes/user-guides/snapshots.md [hunk 4]
```diff
@@ -1,25 +1,8 @@
...
-    ```
-
-1. Deploy driver to Kubernetes Cluster
-
-    ```
-    GCE_PD_SA_DIR=/my/safe/credentials/directory    # Directory to get the service account key
-    GCE_PD_DRIVER_VERSION=alpha                     # Driver version to deploy
-    ./deploy/kubernetes/deploy-driver.sh
-    ```
+### Install Driver with beta snapshot feature as described [here] (https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/docs/kubernetes/user-guides/driver-install.md)
```
Comment (msau42) [inline docs/kubernetes/user-guides/snapshots.md:5]: sorry one more nit. I think you can use a relative url here. That way if someone goes to a tagged version of the docs, it'll go to the page from the same tag.

Comment (saikat-royc) [reply docs/kubernetes/user-guides/snapshots.md:5]: Can you please elaborate how the relative url would look like here?

Comment (msau42) [reply docs/kubernetes/user-guides/snapshots.md:5]: The README has some examples: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blame/master/README.md#L90

Comment (saikat-royc) [reply docs/kubernetes/user-guides/snapshots.md:5]: Thanks will take a look

File: docs/kubernetes/user-guides/snapshots.md [hunk 5]
```diff
@@ -1,25 +1,8 @@
...
-    ```
-
-1. Deploy driver to Kubernetes Cluster
-
-    ```
-    GCE_PD_SA_DIR=/my/safe/credentials/directory    # Directory to get the service account key
-    GCE_PD_DRIVER_VERSION=alpha                     # Driver version to deploy
-    ./deploy/kubernetes/deploy-driver.sh
-    ```
+### Install Driver with beta snapshot feature as described [here] (driver-install.md)
```
Comment (msau42) [inline docs/kubernetes/user-guides/snapshots.md:5]: When I preview this file, the link isn't showing up properly. I think you need to remove the space between "]" and "("

Comment (saikat-royc) [reply docs/kubernetes/user-guides/snapshots.md]: Ah. Thanks for the preview tip. Fixed it.

**Comment by msau42** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**